### PR TITLE
Sumup choice modal on failure

### LINKED ISSUE #766: Update transaction when Sumup fails
When a sumup transaction via the app fails, we should remove the transaction or maybe retry it. It can be checked using the `smp-status` parameter in the callback. 

References:
https://github.com/sumup/sumup-ios-url-scheme/blob/master/README.md
https://github.com/sumup/sumup-android-api/blob/master/README.md

### PR DESCRIPTION
**Pull request by wilco375**
Fixes #766 

Adds a modal that gives the user a choice to retry a Sumup transaction on failure, or delete the order.

image

### CODE REVIEW

File: app/javascript/packs/order_screen.js [hunk 1]
```diff
@@ -160,7 +160,6 @@ document.addEventListener('turbolinks:load', () => {
 
             if (user) {
               this.$set(this.users, this.users.indexOf(this.selectedUser), response.body.user);
-              this.$emit('updateusers');
```
Comment (wilco375) [inline app/javascript/packs/order_screen.js:163]: These events don't seem to be used anywhere so I removed them.

**Comment by DrumsnChocolate** [comment]
Is there a way to test this locally?

**Comment by wilco375** [comment]
> Is there a way to test this locally?

There is, you can go to a url such as http://localhost:5000/activities/<activity_id>/sumup_callback?smp-status=failed&foreign-tx-id=<order_id>

**Comment by DrumsnChocolate** [comment]
okay I've tested this a bit, and I've found a mildly interesting oversight: if you visit http://localhost:5000/activities/<activity_id>/sumup_callback?smp-status=failed&foreign-tx-id=<order_id> for an activity that has been 'locked', and you press "bestelling verwijderen" on the modal that pops up, the deletion doesn't actually succeed (the transaction is unaffected) but the flash notification tells you that you successfully deleted the transaction. Not really a scenario that will occur in reality, but to me it is a hint that the response is not actually verified. Is that correct?

**Comment by wilco375** [comment]
> okay I've tested this a bit using your suggestion (clever way of testing btw), and I've found a mildly interesting oversight: if you visit http://localhost:5000/activities/<activity_id>/sumup_callback?smp-status=failed&foreign-tx-id=<order_id> for an activity that has been 'locked', and you press "bestelling verwijderen" on the modal that pops up, the deletion doesn't actually succeed (the transaction is unaffected) but the flash notification tells you that you successfully deleted the transaction. Not really a scenario that will occur in reality, but to me it is a hint that the response is not actually verified. Is that correct?

Interesting thought. It looks like this was actually a backend error, and not a frontend one. The endpoint still returned a 200 even if the activity was locked and nothing was updated. I fixed that, and it also shows a flash notification with an error now.

**Review by DrumsnChocolate — APPROVED** [review]

**Pull request merged**